olp-cpp-sdk-authentication/include/olp/authentication/TokenProvider.h

```diff
@@ -65,10 +65,32 @@ class TokenProvider {
       : impl_(std::make_shared<TokenProviderImpl>(
             std::move(settings), std::chrono::seconds(MinimumValidity))) {}
 
-  // Default copyable, default movable.
+  /**
+   * @brief A default copy constructor.
+   *
+   * @param other The `TokenProvider` instance.
+   */
```

Comment:
```suggestion
  /// A default copy constructor.
```

Same below and for all other classes.
The following constructors and operators I would only comment with a short one liner:
- default constructor
- destructor (if even needed as it is pretty obvious)
- default copy constructor
- default move constructor
- default copy assignment operator
- default move assignment operator